### ISSUE TITLE
Training from Scratch vs. Provided Weights

### ISSUE THREAD

**Issue by Xujinming1** [Aradhye2002/EcoDepth#25]
Hello,

Thank you for sharing your work and code! Inspired by it, we want to explore the variations of EcoDepth. Therefore, we are trying to reproduce your work by training the model from scratch. Specifically, we:
- Commented out Line 23 to 28 (checkpoint loading) in train.py.
- Keep `train_from_scratch` in train_config.json as false.
- Dataset: NYUv2

I'm using the following default values:
- lr: 1e-5
- weight decay: 0.1
- batch size: 1
- max depth: 10
- no of classes: 100
- epochs: 25
- Training Environment: pytorch 2.5.0 + cuda 11.8, pytoch lightning 2.5.0

Currently, our validation accuracy is 0.57 as d1, significantly lower than the accuracy from pretrained weighted (d1 0.978).

Is there anything I might have missed in reproducing the training process? Or are there any specific tricks or settings used for training the provided weights that are not mentioned in the documentation?

Looking forward to your guidance.

Jinming

**Comment by Aradhye2002** [comment]
Hi Jinming, thank you for appreciating our work!

The behavior you're observing is expected as you're training the model entirely from scratch, which means both the ViT inside the CIDE module and the stable-diffusion UNet are randomly initialized. In our experiments, we used pretrained checkpoints for these components.

The primary intent behind v2.0.0 is to support inference and evaluation using our released checkpoints. While training is still possible, we only support fine-tuning from these checkpoints, as we aimed to keep the restructured codebase simple without introducing logic to load pretrained components from multiple sources.

If you'd like to reproduce our training results, we recommend using the `v1.0.0` tag, which corresponds to the original codebase used in our experiments. That version supports training from scratch while properly loading the necessary pretrained modules.

Hope this helps!

**Comment by Xujinming1** [comment]
Hi @Aradhye2002,

Following your instructions, d1 gets 0.92 with only 1 epoch, which is much better than what I tried before!

Thanks for your help and prompt reply!

**Comment by Xujinming1** [comment]
Hello @Aradhye2002,

I've met another problem now. For some reason I didn't use v1.0.0 for training straightly. But by modifying v2.0.0 (containing loading v1-5-pruned-emaonly.ckpt and so on), I've reproduced the training on NYUv2 successfully.

However while training KITTI, d1 only gets 0.10 with 1 epoch which is much lower than that on NYUv2. And later d1 maintains a lower value.

Compared to NYUv2 training, I'm using following settings (different with NYUv2):
- eval_crop: garg
- scene: outdoor
- max_depth: 80
- depth_factor: 256
- no of classes: 200 (Actually I find that in v1.0.0, train_kitti.sh still uses no_of_classes with 100. But I tried both of them, getting similar results)
- dataset preprocess: do_kb_crop

Other settings (lr 1e-5, weight_decay 0.1 ...) maintain the same as the version, which I successfully reproduced training on NYUv2. So is there anything I missed or any other tricks used for training on KITTI?

Looking forward to your guidance.

**Comment by Xujinming1** [comment]
Hello @Aradhye2002,

I've met another problem now. For some reason I didn't use v1.0.0 for training straightly. But by modifying v2.0.0 (containing loading v1-5-pruned-emaonly.ckpt and so on), I've reproduced the training on NYUv2 successfully.

However while training KITTI, d1 only gets 0.10 with 1 epoch which is much lower than that on NYUv2. And later d1 maintains a lower value.

Compared to NYUv2 training, I'm using following settings (different with NYUv2):
- eval_crop: garg
- scene: outdoor
- max_depth: 80
- depth_factor: 256
- no of classes: 200 (Actually I find that in v1.0.0, train_kitti.sh still uses no_of_classes with 100. But I tried both of them, getting similar results)
- dataset preprocess: do_kb_crop

Other settings (lr 1e-5, weight_decay 0.1 ...) maintain the same as the version, which I successfully reproduced training on NYUv2. So is there anything I missed or any other tricks used for training on KITTI?

Looking forward to your guidance.

**Comment by Wild-Stephen** [comment]
> Hello [@Aradhye2002](https://github.com/Aradhye2002),
> 
> I've met another problem now. For some reason I didn't use v1.0.0 for training straightly. But by modifying v2.0.0 (containing loading v1-5-pruned-emaonly.ckpt and so on), I've reproduced the training on NYUv2 successfully.
> 
> However while training KITTI, d1 only gets 0.10 with 1 epoch which is much lower than that on NYUv2. And later d1 maintains a lower value.
> 
> Compared to NYUv2 training, I'm using following settings (different with NYUv2):
> 
> * eval_crop: garg
> * scene: outdoor
> * max_depth: 80
> * depth_factor: 256
> * no of classes: 200 (Actually I find that in v1.0.0, train_kitti.sh still uses no_of_classes with 100. But I tried both of them, getting similar results)
> * dataset preprocess: do_kb_crop
> 
> Other settings (lr 1e-5, weight_decay 0.1 ...) maintain the same as the version, which I successfully reproduced training on NYUv2. So is there anything I missed or any other tricks used for training on KITTI?
> 
> Looking forward to your guidance.

Hello, I am also reproducing this project, could you please tell me where I can download kitti dataset, ibims, sunrgbd which are matched with this project filenames? waiting for your kind reply, thank you

**Comment by Aradhye2002** [comment]
> > Hello [@Aradhye2002](https://github.com/Aradhye2002),
> > I've met another problem now. For some reason I didn't use v1.0.0 for training straightly. But by modifying v2.0.0 (containing loading v1-5-pruned-emaonly.ckpt and so on), I've reproduced the training on NYUv2 successfully.
> > However while training KITTI, d1 only gets 0.10 with 1 epoch which is much lower than that on NYUv2. And later d1 maintains a lower value.
> > Compared to NYUv2 training, I'm using following settings (different with NYUv2):
> > 
> > * eval_crop: garg
> > * scene: outdoor
> > * max_depth: 80
> > * depth_factor: 256
> > * no of classes: 200 (Actually I find that in v1.0.0, train_kitti.sh still uses no_of_classes with 100. But I tried both of them, getting similar results)
> > * dataset preprocess: do_kb_crop
> > 
> > Other settings (lr 1e-5, weight_decay 0.1 ...) maintain the same as the version, which I successfully reproduced training on NYUv2. So is there anything I missed or any other tricks used for training on KITTI?
> > Looking forward to your guidance.
> 
> Hello, I am also reproducing this project, could you please tell me where I can download kitti dataset, ibims, sunrgbd which are matched with this project filenames? waiting for your kind reply, thank you

Hi @Wild-Stephen, it would be good to create a new issue as this one has been closed, and the topic of discussion is different from the one in this issue.
